src/lib.rs

```diff
@@ -314,22 +317,41 @@ mod tests {
     }
 
     #[test]
-    fn run_test_iter_mut_wrap() {
-        fn test_iter_mut_wrap(mut b: impl RingBufferExt<i32>) {
+    fn test_iter_mut_wrap() {
+        fn run_test_iter_mut_wrap(mut b: impl RingBufferExt<i32>) {
             b.push(1);
             b.push(2);
             b.push(3);
 
-            let mut i = b.iter_mut();
-            while let Some(el) = i.next() {
-                *el += 1;
+            for i in b.iter_mut() {
+                *i += 1;
+            }
+
+            assert_eq!(vec![3, 4], b.to_vec())
+        }
+
+        run_test_iter_mut_wrap(AllocRingBuffer::with_capacity(2));
+        run_test_iter_mut_wrap(ConstGenericRingBuffer::<i32, 2>::new());
+    }
+
+    #[test]
+    fn test_iter_mut_miri_fail() {
```

Comment:
This is the most important function. Without all the unsafe functions (like `ptr_len` and `ptr_capacity` and `ptr_get_mut` which *only* use raw pointers, no references anywhere!!! this is safe. However, as soon as we construc a mutable reference anywhere, miri detects it as aliasing (even though we're technically not able to use both at the same time). This is similar to what `std::vec::Vec` does, however it uses raw pointers everywhere anyway. I think all the unsafe code is very thoroughly tested, and miri passes so I'm quite confident about its correctness.